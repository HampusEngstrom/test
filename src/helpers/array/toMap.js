export const toMap = (arr, key) => {
  const result = {};
  for (let i = 0; i < arr.length; i++) {
    const item = arr[i];
    if (item.hasOwnProperty(key)) {
      const value = item[key];
      result[value] = item;
    }
  }
  return result;
};
