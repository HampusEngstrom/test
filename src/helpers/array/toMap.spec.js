import { toMap } from './toMap';
{
}
describe('test toMap', () => {
  it('returns map', () => {
    const arr = [{ id: '1' }, { id: '2' }, { id: '3' }, { id: '4' }];
    const result = toMap(arr, 'id');
    expect(result).toMatchSnapshot();
  });

  it('returns map for duplicate', () => {
    const arr = [
      { id: '1', name: 'hej1' },
      { id: '2', name: 'hej2' },
      { id: '3', name: 'hej3' },
      { id: '4', name: 'hej4' },
      { id: '4', name: 'hej5' },
    ];
    const result = toMap(arr, 'id');
    expect(result).toMatchSnapshot();
  });
});
