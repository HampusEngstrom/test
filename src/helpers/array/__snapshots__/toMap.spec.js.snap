// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test toMap returns map 1`] = `
Object {
  "1": Object {
    "id": "1",
  },
  "2": Object {
    "id": "2",
  },
  "3": Object {
    "id": "3",
  },
  "4": Object {
    "id": "4",
  },
}
`;

exports[`test toMap returns map for duplicate 1`] = `
Object {
  "1": Object {
    "id": "1",
    "name": "hej1",
  },
  "2": Object {
    "id": "2",
    "name": "hej2",
  },
  "3": Object {
    "id": "3",
    "name": "hej3",
  },
  "4": Object {
    "id": "4",
    "name": "hej5",
  },
}
`;
